packages/wonder-blocks-core/src/components/__tests__/unique-id-provider.test.js

```diff
@@ -51,26 +50,38 @@ describe("UniqueIDProvider", () => {
             expect(children.mock.calls[0][0]).toBeInstanceOf(UniqueIDFactory);
         });
 
-        test("all renders get same unique id factory", () => {
+        test("all renders get same unique id factory", async () => {
             // Arrange
             const children = jest.fn(() => <View />);
             const nodes = (
                 <UniqueIDProvider mockOnFirstRender={false}>
                     {children}
                 </UniqueIDProvider>
             );
-            const wrapper = mount(nodes);
+            const {rerender} = render(nodes);
 
             // Act
-            wrapper.instance().forceUpdate();
+            rerender(nodes);
 
             // Assert
             // Check our forced render worked and we rendered three times.
-            expect(children).toHaveBeenCalledTimes(2);
-            // Check the first render gets a UniqueIDFactory instance.
-            expect(children.mock.calls[0][0]).toBeInstanceOf(UniqueIDFactory);
-            // Check the second render gets the same UniqueIDFactory instance.
-            expect(children.mock.calls[1][0]).toBe(children.mock.calls[1][0]);
+            waitFor(() => {
+                expect(children).toHaveBeenCalledTimes(2);
+            });
+
+            waitFor(() => {
+                // Check the first render gets a UniqueIDFactory instance.
+                expect(children.mock.calls[0][0]).toBeInstanceOf(
+                    UniqueIDFactory,
+                );
+            });
+
+            waitFor(() => {
+                // Check the second render gets the same UniqueIDFactory instance.
+                expect(children.mock.calls[1][0]).toBe(
+                    children.mock.calls[1][0],
+                );
+            });
```

Comment:
**question**: Shouldn't this be a single `waitFor` containing all the expectations? Also, doesn't `waitFor` need to be `await`'d?

Reply:
Oof thanks for pointing this out! I found out that there was an issue with `rerender` that was leading to some false positives.

I made a refactor to these tests so it now works consistently + we don't have to use `waitFor` at all.

The problem was that I was using the reference to the nodes, instead of using a reference to a component instance. This was causing `rerender` not to re-render at all.

BEFORE:

```
const nodes = (<Some><Tree/></Some>);
const {rerender} = render(nodes);
rerender(nodes); // this doesn't re-render internally.
```

AFTER:
```
const UnderTest = () => (<Some><Tree/></Some>);
const {rerender} = render(<UnderTest />);
rerender(<UnderTest />); // this now forces a re-render as expected.
```